cpp/tests/framework/Engine.test.cpp

```
@@ -2,21 +2,59 @@
 #include <atomic>
 #include <catch2/catch_all.hpp>
 #include <chrono>
+#include <netinet/in.h>
+#include <sys/socket.h>
 #include <thread>
 
 namespace robotick::test
 {
+	namespace
+	{
+		struct ThreadJoiner
```

Clang (14.0.6)

[warning] class 'ThreadJoiner' defines a non-default destructor but does not define a copy constructor, a copy assignment operator, a move constructor or a move assignment operator

(cppcoreguidelines-special-member-functions)